lib/matplotlib/stackplot.py

```diff
@@ -68,7 +68,8 @@ def stackplot(axes, x, *args, **kwargs):
 
     baseline = kwargs.pop('baseline', 'zero')
     # Assume data passed has not been 'stacked', so stack it here.
-    stack = np.cumsum(y, axis=0)
+    # We'll need a float buffer for the upcoming calculations.
+    stack = np.cumsum(y, axis=0).astype(np.promote_types(y.dtype, np.float32))
```

Comment:
No idea whether it is optimized to avoid the memory allocation but at least it could do so in theory, whereas `astype` certainly cannot... so I changed it accordingly.